core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphqlActionBuilder.kt

```diff
@@ -0,0 +1,127 @@
+
+
+class GraphqlActionBuilder {
+
+    companion object {
+        private val log: Logger = LoggerFactory.getLogger(GraphqlActionBuilder::class.java)
+        private val idGenerator = AtomicInteger()
+
+        fun addActionsFromSchema(schema: TypeDefinitionRegistry,
+                                 actionCluster: MutableMap<String, Action>) {
+
+            actionCluster.clear()
+
+            val rootOperations = mutableListOf<String>("Mutation", "Query")
+
+            rootOperations.forEach{
+                val obj = schema.getType(it).get() as ObjectTypeDefinition
+                val operationType = it
+                val operationList = obj.fieldDefinitions.forEach{
+
+                    val params = extractParams(it, schema)
+
+                    val action = GraphqlCallAction("${it.name}${idGenerator.incrementAndGet()}", it.name, params, OperationType.valueOf(operationType))
+
+                    actionCluster.put(action.getName(), action)
+                }
+            }
+        }
+
+
+        private fun extractParams(operation: FieldDefinition, schema: TypeDefinitionRegistry): MutableList<GraphqlParam> {
+                    val params: MutableList<GraphqlParam> = mutableListOf()
+                    val name = operation.name
+
+                    operation.inputValueDefinitions.forEach{
+                        val inputName = it.name
+
+                        val type: TypeName
+                        when {
+                            it.type is NonNullType ->
+                                type = (it.type as NonNullType).type as TypeName
+                            it.type is TypeName ->
+                                type = it.type as TypeName
+//                            it.type is ListType -> //TODO: should handle listType
+//                                type = null
+
+                            else -> throw IllegalStateException("Invalid type supplied")
+                        }
+
+
+                        var gene = getGene(inputName, type, schema)
+                        params.add(GraphqlParam(name, gene))
+                    }
+
+            return params
+        }
+
+        private fun getGene(
+                name: String,
+                type: TypeName,
+                schema: TypeDefinitionRegistry
+        ) :Gene {
+            val typeName = type.name
+
+            /*
+            https://graphql.org/learn/schema/
+
+            scalar types:
+            Int  A signed 32‐bit integer
+            Float
+            String
+            Boolean
+            ID      The ID type is serialized in the same way as a String
+
+             */
+            when (typeName) {
+                "Int" -> return IntegerGene(name)
+                "Float" -> return FloatGene(name)
+                "String" -> return StringGene(name)
+                "Boolean" -> return BooleanGene(name)
+                "ID" -> return StringGene(name) //according to the reference we can treat ID as String
```

Comment:
are the type name case-sensitive?

Reply:
Well I think primitive types are case-sensitive.

Reply:
yep, it seems you are right ;) 
https://github.com/graphql/graphql-spec/issues/33